packages/repository/test/acceptance/has-one.relation.acceptance.ts

```diff
@@ -0,0 +1,165 @@
+import {CustomerRepository, AddressRepository} from '../fixtures/repositories';
+import {Where} from '../..';
+
+describe('hasOne relation', () => {
+  // Given a Customer and Address models - see definitions at the bottom
+
+  let app: ApplicationWithRepositories;
+  let controller: CustomerController;
+  let customerRepo: CustomerRepository;
+  let addressRepo: AddressRepository;
+  let existingCustomerId: number;
+
+  before(givenApplicationWithMemoryDB);
+  before(givenBoundCrudRepositoriesForCustomerAndAddress);
+  before(givenCustomerController);
+
+  beforeEach(async () => {
+    await addressRepo.deleteAll();
+  });
+
+  beforeEach(async () => {
+    existingCustomerId = (await givenPersistedCustomerInstance()).id;
+  });
+
+  it('can create an instance of the related model', async () => {
+    const address = await controller.createCustomerAddress(existingCustomerId, {
+      street: '123 test avenue',
+    });
+    expect(address.toObject()).to.containDeep({
+      customerId: existingCustomerId,
+      street: '123 test avenue',
+    });
+
+    const persisted = await addressRepo.findById(address.customerId);
+    expect(persisted.toObject()).to.deepEqual(address.toObject());
+  });
+
+  it("doesn't allow to create related model instance twice", async () => {
+    const address = await controller.createCustomerAddress(existingCustomerId, {
+      street: '123 test avenue',
+    });
+    expect(
+      controller.createCustomerAddress(existingCustomerId, {
+        street: '456 test street',
+      }),
+    ).to.be.rejectedWith(/Duplicate entry for Address.customerId/);
+    expect(address.toObject()).to.containDeep({
+      customerId: existingCustomerId,
+      street: '123 test avenue',
+    });
+
+    const persisted = await addressRepo.findById(address.customerId);
+    expect(persisted.toObject()).to.deepEqual(address.toObject());
+  });
+
+  it('can find instance of the related model', async () => {
+    const address = await controller.createCustomerAddress(existingCustomerId, {
+      street: '123 test avenue',
+    });
+    const notMyAddress = await controller.createCustomerAddress(
+      existingCustomerId + 1,
+      {
+        street: '456 test road',
+      },
+    );
+    const foundAddress = await controller.findCustomerAddress(
+      existingCustomerId,
+    );
+    expect(foundAddress).to.containEql(address);
+    expect(foundAddress).to.not.containEql(notMyAddress);
+
+    const persisted = await addressRepo.find({
+      where: {customerId: existingCustomerId},
+    });
+    expect(persisted[0]).to.deepEqual(foundAddress);
+  });
+
+  it('does not allow where filter to find related model instance', async () => {
+    const address = await controller.createCustomerAddress(existingCustomerId, {
+      street: '123 test avenue',
+    });
+
+    const foundAddress = await controller.findCustomerAddressWithFilter(
+      existingCustomerId,
+      {where: {street: '123 test avenue'}},
```

Comment:
The purpose of the `filter` argument here is to accept things like `fields` in it, but it also ignores `where` clauses since there is only one instance (the type for it is `filter?: Exclude<Filter<TargetEntity>, Where<TargetEntity>>`). This test is to ensure that it ignores the `where` object.